### PR TITLE
EVG-15720: Upgrade to newer version of protobuf

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-15720

### CODE REVIEW

File: buildlogger.proto [hunk 1]
```diff
@@ -2,7 +2,7 @@ syntax = "proto3";
 
 package cedar;
 
-option go_package = "gopb";
+option go_package = "./gopb";
```
Comment (Kimchelly) [inline buildlogger.proto:5]: Is this something that's required by the newer version of protobuf?

File: makefile [hunk 2]
```diff
@@ -41,10 +39,10 @@ $(buildDir)/protoc:
 	curl --retry 10 --retry-max-time 60 -L0 https://github.com/protocolbuffers/protobuf/releases/download/v$(protocVersion)/protoc-$(protocVersion)-$(protoOS).zip --output protoc.zip
 	unzip -q protoc.zip -d $(buildDir)/protoc
 	rm -f protoc.zip
-	GOBIN="$(abspath $(buildDir))" $(gobin) install github.com/golang/protobuf/protoc-gen-go@v$(protocGenGoVersion)
+	GOBIN="$(abspath $(buildDir))" $(gobin) install google.golang.org/protobuf/cmd/protoc-gen-go@latest
```
Comment (Kimchelly) [inline makefile:42]: Is it always okay to use the latest version of protoc-gen-go? Wouldn't that require us to also upgrade protoc to be sufficiently up-to-date for this to work?

File: buildlogger.proto (hunk 1 above)
Comment (julianedwards) [reply buildlogger.proto:5]: Yeah, it's really annoying. They really don't want you to build your pb files in the same directory. But it requires either a slash or dot.

File: makefile (hunk 2 above)
Comment (julianedwards) [reply makefile:42]: I think it should be fine, but we should probably be updating both either way. Unfortunately there is no "latest" URL for protoc.

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**